Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -426,37 +562,57 @@ int CruRawReader::processHalfCRU(int iteration)
       checkNoWarn();
     }
     incrementErrors(TRDParsingHalfCRUSumLength);
+    if (mOptions[TRDVerboseErrorsBit]) {
+      LOG(warn) << "Next HalfCRU header says it contains more data than in the rdh payloads! "
+                << totalHalfCRUDataLength32 << " < " << mTotalHBFPayLoad << "-" << mHBFoffset32
+                << " sector:side:endpoint: " << (unsigned int)mFEEID.supermodule << ":"
+                << (unsigned int)mFEEID.side << ":" << (unsigned int)mFEEID.endpoint;
+    }
     mWordsRejected += (mTotalHBFPayLoad / 4) - mHBFoffset32;
     return -2;
   }
-  if (!halfCRUHeaderSanityCheck(mCurrentHalfCRUHeader, mCurrentHalfCRULinkLengths, mCurrentHalfCRULinkErrorFlags)) {
-    if (mMaxWarnPrinted > 0) {
-      LOG(warn) << "HalfCRU header failed sanity check for FEEID with  sector:side:endpoint: " << (unsigned int)mFEEID.supermodule << ":" << (unsigned int)mFEEID.side << ":" << (unsigned int)mFEEID.endpoint;
-      checkNoWarn();
-    }
-    // let incrementErrors catch the undefined values of sector side stack and layer as if not set it will go so zero in the method, however if set, it means this is the second half cru header, and we have the values from the last one we read which
+  if (!halfCRUHeaderSanityCheck(mCurrentHalfCRUHeader, mCurrentHalfCRULinkLengths,
+                                mCurrentHalfCRULinkErrorFlags)) {
+    // let incrementErrors catch the undefined values of sector side stack and layer as if not
+    // set it will go so zero in the method, however if set, it means this is the second half
+    // cru header, and we have the values from the last one we read which
     // *SHOULD* be the same as this halfcruheader.
+    if (mOptions[TRDVerboseErrorsBit]) {
+      LOG(info) << "HalfCRU header failed sanity check for FEEID with  sector:side:endpoint: "
+                << (unsigned int)mFEEID.supermodule << ":" << (unsigned int)mFEEID.side << ":"
+                << (unsigned int)mFEEID.endpoint;
+    }
     incrementErrors(TRDParsingHalfCRUCorrupt);
     mWordsRejected += (mTotalHBFPayLoad / 4) - mHBFoffset32;
     return -2;
   }
 
   //get eventrecord for event we are looking at
-  mIR.bc = mCurrentHalfCRUHeader.BunchCrossing; // correct mIR to have the physics trigger bunchcrossing *NOT* the heartbeat trigger bunch crossing.
+  mIR.bc = mCurrentHalfCRUHeader.BunchCrossing; // correct mIR to have the physics trigger
+                                                // bunchcrossing *NOT* the heartbeat trigger
+                                                // bunch crossing.
```

Comment:
Did you enforce now the maximum line length in the editor? I am not sure if it makes it so much more readable

Reply:
certainly not there.